v2/robotmk/src/bin/agent.rs

```diff
@@ -0,0 +1,139 @@
+use camino::Utf8PathBuf;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use std::env::{var, VarError};
+use std::fs::read_to_string;
+use std::io;
+use std::path::Path;
+use walkdir::{DirEntry, WalkDir};
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub results_directory: Utf8PathBuf,
+}
+
+#[derive(Parser)]
+#[command(about = "Robotmk agent plugin.")]
+struct Args {
+    /// Configuration file path.
+    #[clap(name = "CONFIG_PATH")]
+    pub config_path: Option<Utf8PathBuf>,
+}
+
+#[derive(Serialize)]
+pub struct ConfigError {
+    config_reading_error: String,
+}
+
+#[derive(Serialize)]
+pub struct ConfigFileContent {
+    config_file_content: String,
+}
+
+fn determine_config_path(arg: Option<Utf8PathBuf>) -> Result<Utf8PathBuf, String> {
+    if let Some(p) = arg {
+        return Ok(p);
+    }
+    let config_dir = match var("MK_CONFDIR") {
+        Ok(path) => path,
+        Err(VarError::NotPresent) => "C:\\ProgramData\\checkmk\\agent\\config".to_string(),
+        Err(VarError::NotUnicode(_path)) => return Err("CONFIG_PATH is not utf-8.".into()),
+    };
+    let mut config_dir = Utf8PathBuf::from(config_dir);
+    config_dir.push("robotmk.json");
+    Ok(config_dir)
+}
+
+fn report_config_error(message: String) {
+    let config_error = serde_json::to_string(&ConfigError {
+        config_reading_error: message,
+    })
+    .unwrap();
```

Comment:
I would prefer to return a result here and unwrap in `main`, but I think that's more a question of taste, so no need to change it.